File: 0704_binary_search.swift
// https://leetcode.com/problems/binary-search/

class Solution {
    func search(_ nums: [Int], _ target: Int) -> Int {
        var l = 0
        var r = nums.count - 1

        while l <= r {
            let mid = (l + r) / 2
            let midValue = nums[mid]
            if midValue == target {
                return mid
            } else if target > midValue {
                l = mid + 1
            } else {
                r = mid - 1
            }
        }

        return -1
    }
}

File: 0704_binary_search.kt
// https://leetcode.com/problems/binary-search/description/

class Solution {
    fun search(nums: IntArray, target: Int): Int {
        var l = 0
        var r = nums.size - 1
        while (l <= r) {
            val mid = (l + r) / 2
            val midNum = nums[mid]
            if (target <= midNum) {
                r = mid - 1
            } else {
                l = mid + 1
            }
        }
        if (l < nums.size && nums[l] == target) {
            return l
        } else {
            return -1
        }
    }
}
